### PR TITLE
doc: Split depends installation instructions per arch

### PR DESCRIPTION
**Pull request by maflcko**
The current depends installation instructions fail on bionic with

```
E: Unable to locate package g++-4.8-aarch64-linux-gnu
E: Unable to locate package gcc-4.8-aarch64-linux-gnu
E: Unable to locate package g++-4.8-arm-linux-gnueabihf
E: Unable to locate package gcc-4.8-arm-linux-gnueabihf
```

Also, they fail due to missing dependencies `make automake cmake pkg-config python3`

Fix this by removing the explicit version and splitting them into common instructions and instructions per linux architecture.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Fun fact:

```
git log 000000 -1
error: short SHA1 000000 is ambiguous
hint: The candidates are:
hint:   000000035 commit 2018-06-20 - Obsolete #!/bin/bash shebang
hint:   000000901 commit 2018-09-18 - doc: Split depends installation instructions per arch
fatal: ambiguous argument '000000': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'

**Comment by MarcoFalke** [comment]
Tested that depends compilation (and cross-compilation) works now on debian:9, xenial and bionic.

File: depends/README.md [hunk 1]
```diff
@@ -30,29 +30,38 @@ Common `host-platform-triplets` for cross compilation are:
...
+#### For linux (including i386, ARM) cross compilation:
 
-    sudo apt-get install curl g++-aarch64-linux-gnu g++-4.8-aarch64-linux-gnu gcc-4.8-aarch64-linux-gnu binutils-aarch64-linux-gnu g++-arm-linux-gnueabihf g++-4.8-arm-linux-gnueabihf gcc-4.8-arm-linux-gnueabihf binutils-arm-linux-gnueabihf g++-4.8-multilib gcc-4.8-multilib binutils-gold bsdmainutils
+Common linux dependencies:
+
+    sudo apt-get install make automake cmake curl g++-multilib libtool binutils-gold bsdmainutils pkg-config python3
+
+For linux ARM cross compilation:
+
+    sudo apt-get install g++-aarch64-linux-gnu binutils-aarch64-linux-gnu
```
Comment (arowser) [inline depends/README.md]: line 51 is for aarch64,  I guess you should swap with line 55.

File: depends/README.md [hunk 2]
```diff
@@ -30,29 +30,38 @@ Common `host-platform-triplets` for cross compilation are:
...
---------------------------------------------------
+### Install the required dependencies: Ubuntu & Debian
 
-For macOS cross compilation:
+#### For macOS cross compilation:
 
     sudo apt-get install curl librsvg2-bin libtiff-tools bsdmainutils cmake imagemagick libcap-dev libz-dev libbz2-dev python-setuptools
 
-For Win32/Win64 cross compilation:
+#### For Win32/Win64 cross compilation:
```
Comment (practicalswift) [inline depends/README.md]:
```
2018-09-19 20:32:54 mdl(pr=14264): depends/README.md:39: MD026 Trailing punctuation in header
```

File: depends/README.md [hunk 3]
```diff
@@ -30,29 +30,38 @@ Common `host-platform-triplets` for cross compilation are:
...
 
 - see [build-windows.md](../doc/build-windows.md#cross-compilation-for-ubuntu-and-windows-subsystem-for-linux)
 
-For linux (including i386, ARM) cross compilation:
+#### For linux (including i386, ARM) cross compilation:
```
Comment (practicalswift) [inline depends/README.md]:
```
2018-09-19 20:32:54 mdl(pr=14264): depends/README.md:43: MD026 Trailing punctuation in header
```

File: depends/README.md [hunk 4]
```diff
@@ -30,29 +30,38 @@ Common `host-platform-triplets` for cross compilation are:
 
 No other options are needed, the paths are automatically configured.
 
-Install the required dependencies: Ubuntu & Debian
---------------------------------------------------
+### Install the required dependencies: Ubuntu & Debian
 
-For macOS cross compilation:
+#### For macOS cross compilation:
```
Comment (practicalswift) [inline depends/README.md]:
```
2018-09-19 20:32:54 mdl(pr=14264): depends/README.md:47: MD026 Trailing punctuation in header
```

**Comment by MarcoFalke** [comment]
@Sjors I've stolen this from you. Mind to take a look here if it looks right?

**Comment by Sjors** [comment]
It looks OK to me, but only way to know for sure is to try on a fresh machine, which I haven't gotten around to.

**Comment by MarcoFalke** [comment]
I am going to merge, since the current instructions are broken and I tried this on several fresh machines.

**Pull request merged**